pkg/block/blocktest/adapter.go

```diff
@@ -0,0 +1,311 @@
+package blocktest
+
+import (
+	"context"
+	"io"
+	"net/url"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/go-test/deep"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/block"
+	"github.com/treeverse/lakefs/pkg/ingest/store"
+)
+
+func TestAdapter(t *testing.T, adapter block.Adapter, storageNamespace, externalPath string) {
```

Comment:
Maybe I'm missing something - but this is a test function

Reply:
Not really: it's a helper function for writing test functions as described in [package testing](https://pkg.go.dev/testing).

Final counterpoint from me on this topic: suppose TestAdapter included reading the config  and therefore its callers would set all values from the environment.  It would need to have signature `TestAdapter(t *testing.T)` -- and then package testing would try to run it!  In that case you would need to rename the func.  So naming a function `Test*` is confusing when you expect it _not_ to run automagically.